builder/source/git.go

```diff
@@ -77,43 +77,33 @@ func (g *GitSource) submodules() error {
 	return cmd.Run()
 }
 
-// clone shallow clones an upstream git repository to the local disk.
-func clone(uri, path, ref string) error {
-	var cmd *exec.Cmd
-
-	// Check if the reference is a commit
-	if len(ref) == 40 {
-		// Init a new repository at the checkout path
-		initCmd := exec.Command("git", "init", path)
-
-		if err := initCmd.Run(); err != nil {
-			return err
-		}
+// For some reason git blobless clones create pack files with 600 permissions. These break future operations
+// as those files cannot be read by non-root users. Fix those permissions so things work as they should
```

Comment:
You're missing a period here:

```suggestion
// as those files cannot be read by non-root users. Fix those permissions so things work as they should.
```